application/src/main/java/org/togetherjava/tjbot/commands/moderation/AuditCommand.java

```diff
@@ -0,0 +1,348 @@
+     *
+     * @param event the event of the command
+     */
+    private void handleBanSubCommand(@NotNull CommandInteraction event) {
+        OptionMapping banUserOption =
+                Objects.requireNonNull(event.getOption(BAN_USER_OPTION), OPTION_IS_NULL);
+        User target = banUserOption.getAsUser();
+        Guild guild = Objects.requireNonNull(event.getGuild(), GUILD_IS_NULL);
+
+        Member bot = guild.getSelfMember();
+        Member author = Objects.requireNonNull(event.getMember(), OPTION_IS_NULL);
+
+        if (!handleChecks(bot, author, guild, event, Permission.BAN_MEMBERS, hasBanRequiredRole)) {
+            return;
+        }
+
+        Long guildId = guild.getIdLong();
+        Long userId = target.getIdLong();
+        Optional<String> banReason = database.read(context -> {
+            try (var select = context.selectFrom(BanSystem.BAN_SYSTEM)) {
+                return Optional
+                    .ofNullable(select.where(BanSystem.BAN_SYSTEM.USERID.eq(userId)
+                        .and(BanSystem.BAN_SYSTEM.GUILD_ID.eq(guildId))).fetchOne())
+                    .map(BanSystemRecord::getBanReason);
+            }
+        });
+
+        Optional<Long> banAuthorId = database.read(context -> {
+            try (var select = context.selectFrom(BanSystem.BAN_SYSTEM)) {
+                return Optional
+                    .ofNullable(select.where(BanSystem.BAN_SYSTEM.USERID.eq(userId)
+                        .and(BanSystem.BAN_SYSTEM.GUILD_ID.eq(guildId))).fetchOne())
+                    .map(BanSystemRecord::getAuthorId);
+            }
+        });
+
+        try {
+            Optional<Boolean> isBanned = database.read(context -> {
+                try (var select = context.selectFrom(BanSystem.BAN_SYSTEM)) {
+                    return Optional
+                        .ofNullable(select.where(BanSystem.BAN_SYSTEM.USERID.eq(userId)
+                            .and(BanSystem.BAN_SYSTEM.GUILD_ID.eq(guildId))).fetchOne())
+                        .map(BanSystemRecord::getIsBanned);
+                }
+            });
+            if (!noValueFound(isBanned, target, BANNED, event)
+                    && !noValueFound(banReason, target, BANNED, event)
+                    && !noValueFound(banAuthorId, target, BANNED, event)) {
+                return;
+            }
+
+            event
+                .replyEmbeds(new EmbedBuilder().setTitle("Null")
+                    .setDescription("The user " + target.getAsTag() + " was banned for the reason"
+                            + banReason + " by " + banAuthorId)
+                    .setColor(Color.decode(COLOR_CODE))
+                    .build())
+                .setEphemeral(true)
+                .queue();
+        } catch (Exception exception) {
+            logger.error("Failed to check if the user is banned", exception);
+            replyEphemeral("Failed to check if the user is banned", event);
+        }
+    }
```

Comment:
I will skip reviewing the rest of this for now since there are so many design flaws that will change the code entirely anyways.